From 5f7057974cc62303647c9ca73ecdee51c19a07cf Mon Sep 17 00:00:00 2001
From: liangtian <liangtian13@huawei.com>
Date: Wed, 27 Apr 2022 18:48:32 +0800
Subject: [PATCH 2940/4585] [Backport] x86: KVM: Fixed the bug that WAITmax
 cannot be updated in real time

virt inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I53PTV?from=project-issue

Reference: https://gitee.com/openeuler/kernel/commit/414a578b07f9040de2bf9be340bd6e28326ca788

--------------------------------

Since the reset function is in kvm_intel module instead of kvm
module, the attribute weak function in kvm_main.c could not be found, which
would cause st_max in X86 never be refreshed.
The solution is to define the reset function in x86.c under the kvm module.

Signed-off-by: liangtian <liangtian13@huawei.com>
Reviewed-by: Keqian Zhu <zhukeqian1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 arch/x86/kvm/vmx/vmx.c | 5 -----
 arch/x86/kvm/x86.c     | 5 +++++
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/x86/kvm/vmx/vmx.c b/arch/x86/kvm/vmx/vmx.c
index 68dd25be16aa..79889d27aa5b 100644
--- a/arch/x86/kvm/vmx/vmx.c
+++ b/arch/x86/kvm/vmx/vmx.c
@@ -424,11 +424,6 @@ static u32 vmx_segment_access_rights(struct kvm_segment *var);
 static __always_inline void vmx_disable_intercept_for_msr(struct kvm_vcpu *vcpu,
 							  u32 msr, int type);
 
-void kvm_arch_vcpu_stat_reset(struct kvm_vcpu_stat *vcpu_stat)
-{
-	vcpu_stat->st_max = 0;
-}
-
 void vmx_vmexit(void);
 
 #define vmx_insn_failed(fmt...)		\
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index ddddf8ac2111..cd966a2a3ece 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -11499,6 +11499,11 @@ int kvm_handle_invpcid(struct kvm_vcpu *vcpu, unsigned long type, gva_t gva)
 }
 EXPORT_SYMBOL_GPL(kvm_handle_invpcid);
 
+void kvm_arch_vcpu_stat_reset(struct kvm_vcpu_stat *vcpu_stat)
+{
+	vcpu_stat->st_max = 0;
+}
+
 EXPORT_TRACEPOINT_SYMBOL_GPL(kvm_exit);
 EXPORT_TRACEPOINT_SYMBOL_GPL(kvm_fast_mmio);
 EXPORT_TRACEPOINT_SYMBOL_GPL(kvm_inj_virq);
-- 
2.17.1

